controllers/globals.go

```diff
@@ -0,0 +1,5 @@
+package controllers
+
+// Currently we only have 1 written and 1 reader so there is no reason to
+// add mutexts. If writters are added, we need to make sure to add syncronization.
+var kernelToOS = map[string]string{}
```

Comment:
I think it would be more convenient to map the kernel version to the DTK image here.
This likely means that we need to watch both nodes and the DTK ImageStream to react on updates on both. Sounds like a good reason to move the new code from the node-kernel reconciler to a separate one, IMHO.

Reply:
@qbarrand why do we need to watch the image stream? We can get its data once we really need to process the build.

Reply:
If we watch it, we only have to find the image name for a given OS version once - vs each time we need to create a build.

Reply:
that can easily be mitigated with caching, instead of additional controller...

Reply:
I don't see any downside in having that feature in another controller - clear separation of concerns. The Node informer would be shared together with its cache, so the impact on memory utilization would be limited to the ImageStream watcher.

Reply:
To me, the real value of mapping `kernel-->DTK` is that it will simplify the hub-spoke flow as if a spoke send its kernels and DTKs (instead of kernel+osImages) then we don't need to worry about that specific ocImages not being present in the hub's imagestream.